### ISSUE TITLE
Fail to reproduce results

### ISSUE THREAD

**Issue by JulienBrochier** [oliverguhr/spelling#3]
I'm trying to reproduce results of the English model before making a French one.
Loss stabilizes around 2.0 after 0.2 epochs, slowly decreasing to 1.7.

I'm using provided data and the script generate_dataset.py for dataset generation.

**Training parameters** are default ones, except for batch sizes that I had to decrease from 4 to 2:
- learning_rate: 0.0003
- train_batch_size: 2
- eval_batch_size: 2
- seed: 42
- gradient_accumulation_steps: 8
- total_train_batch_size: 16
- optimizer: Adam with betas=(0.9,0.999) and epsilon=1e-08
- lr_scheduler_type: linear
- num_epochs: 1.0

**Framework versions:**

- Transformers 4.24.0
- Pytorch 1.10.0+cu113
- Datasets 2.6.1
- Tokenizers 0.13.2

I had to manually download model and replace name by path inside train_bart_model.sh.
Model and tokenizer configurations seem to be well retrieved. I've checked it by comparing files config.json and tokenizer_config.json of our models, generated automatically after training.
The only change is my transformers' version being "4.24.0" instead of "4.19.0.dev0".

I had to convert some files to txt format to share them.
[README.md](https://github.com/oliverguhr/spelling/files/10085878/README.md)
[train_bart_model.txt](https://github.com/oliverguhr/spelling/files/10085917/train_bart_model.txt)
[config.txt](https://github.com/oliverguhr/spelling/files/10085892/config.txt)
[tokenizer_config.txt](https://github.com/oliverguhr/spelling/files/10085893/tokenizer_config.txt)

Are there things I'm doing wrong?

**Comment by oliverguhr** [comment]
Mhhh being forced to download the model manually is strange. That should not happen and might be the root cause of the bug.

**Comment by oliverguhr** [comment]
Mhhh being forced to download the model manually is strange. That should not happen and might be the root cause of the bug. 
The only thing I can see right now is, that you reduced the batch size but not increased the gradient_accumulation_steps. To keep the batch equivalent, you can reduce the batch size to 2 and increase gradient accumulation steps to 16.

However, I am running a training with your exact parameters and following versions. I post an update as soon 
as possible. 

    Transformers 4.24.0
    Pytorch  1.13.0 CUDA 11.7
    Datasets 2.7.1
    Tokenizers 0.13.2

**Comment by oliverguhr** [comment]
I cant reproduce your bug. After 0.1 epoch the loss is at about 0.3. 

I submitted my training files, wiki.en.train.csv and wiki.en.test.csv - can you compare them with your files an rerun the training with these files?

**Comment by JulienBrochier** [comment]
Thanks!
I also think that the problem comes from model download. I did it manually because of ssl certificate verification error.   
I will try other ways and with your training files too.
